pkg/kernel/networkservice/connectioncontextkernel/ipcontext/ipaddress/common.go

```diff
@@ -96,6 +96,11 @@ func create(ctx context.Context, conn *networkservice.Connection, isClient bool)
 		if err := netlink.AddrSubscribeAt(targetNetNS, ch, done); err != nil {
 			return errors.Wrapf(err, "failed to subscribe for interface address updates")
 		}
+		defer func() {
+			// `ch` should be fully read after the `done` close to prevent goroutine leak in `netlink.AddrSubscribeAt`
+			for range ch {
```

Comment:
shouldn't `waitForIPNets` method to run in a loop until `ch` is closed instead of reading `ch` values in a stretch ?
```
		case update, ok := <-ch:
			if !ok {
				return nil
			}
			....
```

Reply:
It is not actually `return nil`, it is `return some_err`, because we didn't receive events for all `ipNets`. This behavior is currently actually reached by `ctx.Done()` check, but checking this explicitly is much more better.
Fixed.

Reply:
@Bolodya1997 Can we add select on done channel for waitForIPNets?